Feature/152798633/jwt signup signin

daddychukz opened the pull request:
#### What does this PR do?
it enables user receive JWT upon successful sign-up and sign-in

 #### Description of Task to be completed?
1) User should receive a token upon sign-up
2) User should receive a token upon sign-in

 ### How should this be manually tested?
1) Sign-up within the app
2) Open console on web browser
3) Click on Application tab
4) select Local storage

 #### Any background context you want to provide?
Nil

 #### What are the relevant pivotal tracker stories?
#152798635 Users should receive a JWT on successful signup
#152798633 Users should receive a JWT on successful login

 #### Screenshots (if appropriate)
<img width="1437" alt="screen shot 2017-11-28 at 1 27 55 pm" src="https://user-images.githubusercontent.com/30434186/33321442-02b036a4-d446-11e7-9a6d-277f0ea6dce1.png">

 #### Questions:```

daddychukz committed:
feature(jwt-signup-signin): remove alert message box [Finishes #152798633]

sachungo commented:
@daddychukz houndCI hasn't enabled [Eslint by default](https://houndci.com/configuration)
Can you create `.hound.yml` and add the following?
```yml
fail_on_violations: true
eslint:
  enabled: true
  config_file: .eslintrc.json

sachungo commented on app/controllers/index.js:
```diff
@@ -17,7 +17,7 @@ exports.play = function(req, res) {
+  res.render('index', {
+    user: req.user ? JSON.stringify(req.user) : "null"
+  });
 };
```
Add new line

sachungo commented on app/controllers/users.js:
```diff
@@ -132,21 +194,21 @@ exports.addDonation = function(req, res) {
-        }
-      });
+          if (!duplicate) {
+            console.log('Validated donation');
```
Add a `TODO`comment to remember  to remove the console statements

sachungo commented on app/views/layouts/default.jade:
```diff
@@ -1,4 +1,4 @@
-!!! 5
+doctype html
```
👍 More understandable

sachungo commented on config/socket/socket.js:
```diff
@@ -16,217 +16,217 @@ module.exports = function(io) {
-    socket.on('disconnect', function(){
-      console.log('Rooms on Disconnect ', io.sockets.manager.rooms);
+    socket.on('disconnect', function() {
+      // console.log('Rooms on Disconnect ', io.sockets.manager.rooms);
```
Instead of commenting the line, just delete it

sachungo commented on public/js/controllers/index.js:
```diff
@@ -15,10 +15,55 @@ angular.module('mean.system')
+      $http.post('/users', userData)
+        .then((response) => {
+          $scope.alert = `${response.data.message} You will be redirected after few minutes`;
+          window.localStorage.setItem('token', response.data.token);
```
Consider using sessionStorage instead of localStorage

sachungo commented on public/js/controllers/index.js:
```diff
@@ -15,10 +15,55 @@ angular.module('mean.system')
+        }, (response) => {
+          $scope.alert = response.data.message;
+        });
+    }
```
Should be addressed

sachungo commented on test/user/model.js:
```diff
@@ -2,46 +2,46 @@
+      done();
     });
+  });
 });
```
Add new line

sachungo reviewed (changes requested):
Left some minor feedback with regards to console statements and new line.

👍 for fixing indentations

I haven't tested it yet as the staging deploy doesn't work

daddychukz committed:
feature(jwt-signup-signin):
- Implement feedback
- Add newline to line-endings
[fixes #152798633]

daddychukz committed:
feature(jwt-signup-signin):
- Refactor gulp file
[fixes #152798633]

daddychukz committed:
feature(jwt-signup-signin):
- Move packages to dependency
[finishes #152798633]

daddychukz committed:
move package

daddychukz committed:
feature(jwt-signup-signin):
- implement feedback on naming convention
- set mongoClient to true
[fixes #152798633]

daddychukz committed:
Merge branch 'staging' into feature/152798633/jwt-signup-signin

sachungo reviewed (approved):
Feedback addressed and tested locally
